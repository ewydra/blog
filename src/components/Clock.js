import React, { Component } from "react";
import { format } from "date-fns";

class Clock extends Component {
  state = {
    timestamp: Date.now()
  }

  t = null;

  componentDidMount() {
    this.t = setInterval(() => this.setState({ timestamp: Date.now() }), 1000);
  }

  componentWillUnmount() {
    clearInterval(this.t)
  }

  render() {
    const formattedHour = format(this.state.timestamp, "H:mm:ss");
    return (
      <div className="clock">{formattedHour}</div>
    )
  }
}

export default Clock;
