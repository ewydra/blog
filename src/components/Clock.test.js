import React from "react";
import MockDate from "mockdate";
import { shallow } from "enzyme";
import Clock from "./Clock";

jest.useFakeTimers();

describe("Clock", () => {
  it("should update clock every second", () => {
    MockDate.set(1597862770000);
    
    const component = shallow(<Clock />);

    expect(component.state("timestamp")).toEqual(1597862770000);

    MockDate.set(1597862771000);
    jest.advanceTimersByTime(1000);
    expect(component.state("timestamp")).toEqual(1597862771000);

    MockDate.set(1597862774000);
    jest.advanceTimersByTime(3000);
    expect(component.state("timestamp")).toEqual(1597862774000);
    expect(component).toMatchSnapshot();
  })
})
