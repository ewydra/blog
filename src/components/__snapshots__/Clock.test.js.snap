// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock should update clock every second 1`] = `
<div
  className="clock"
>
  18:46:14
</div>
`;
